rcv/nes56_cv_manager.py

```diff
@@ -0,0 +1,150 @@
+            self._client_socket.close()
+        if self._cap:
+            self._cap.release()
+        if self._show:
+            cv2.destroyAllWindows()
+
+    def init_video_source(self):
+        logging.debug("Enetered Nes56CvManager.init_video_source()")
+        self._video_source_type, self._video_source = self._conf.get('rcv_server', 'video_source').split(':')
+        print("video_source_type = {}".format(self._video_source_type))
+        print("video_source = {}".format(self._video_source))
+        if self._video_source_type == 'camera':
+            self._video_source = int(self._video_source)
+            self._height = float(self._conf.get('rcv_server', 'height'))
+            self._width = float(self._conf.get('rcv_server', 'width'))
+            self._camera_name = self._conf.get('rcv_server', 'camera_name')
+            self._fps = float(self._conf.get('rcv_server', 'fps'))
+        elif self._video_source_type == 'video':
+            self._camera_name = os.path.basename(self._video_source)
+        else:
+            raise ValueError("Unsupported video_source - '{}'".format(self._video_source_type))
+        self._cap = cv2.VideoCapture(self._video_source)
+        if self._video_source_type == 'camera':
+            self._cap.set(cv2.CAP_PROP_FRAME_WIDTH, self._width)
+            self._cap.set(cv2.CAP_PROP_FRAME_HEIGHT, self._height)
+            self._cap.set(cv2.CAP_PROP_FPS, self._fps)
+
+    def is_capture_opened(self):
+        return self._cap.isOpened()
+
+    def connect_to_roborio(self):
+        logging.debug("entered Nes56CvManager.connect_to_roborio")
+        self._client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._client_socket.settimeout(self._socket_timeout)
+        err = None
+        try:
+            err = self._client_socket.connect_ex((self._roborio_ip,
+                                                  self._roborio_port))
+            if err != 0:
+                logging.error("couldn't connect "+ str(err))
+                self._client_socket.close()
+                self._client_socket = None
+            else:
+                logging.debug("connected")
+        except:
+            logging.exception("Exception", exc_info=1)
+
+    def send_data_to_roborio(self, data):
+        logging.debug("entered Nes56CvManager.send_data_to_roborio")
+        try:
+            self._client_socket.sendall(data)
+            logging.debug("sent succesfully {}".format(data))
+        except socket.error as e:
+            logging.error("error in sending " + str(e))
+            self._client_socket.close()
+            self._client_socket = None
+
+
+    def get_next_frame(self):
+        ret, frame = self._cap.read()
+        return (ret, frame)
+
+    def analyze_frame(self, frame):
+        return self._frame_handler.handle_frame(frame,
+                        self._camera_height,
+                        self._y_leaning_angle,
+                        self._x_turning_angle)
```

Comment:
if you remove the camera height y leaning angle and x turning angle you shouldn't give them here because FrameHandler should only get frame if we won't it to be logical and to use this file with another logic

Reply:
O.K. you convinced me